src/epoch.rs

```diff
@@ -2226,6 +2226,82 @@ impl Epoch {
         me
     }
 
+    /// Builds an Epoch from given `wk` week counter into the desired Time scale.
+    /// `ns` is the amount of nanoseconds into that week from closest Sunday Midnight.
+    /// This is usually how GNSS receivers describe a GNSS time scale epoch
+    #[cfg(feature = "python")]
+    #[staticmethod]
+    pub fn from_timeofweek(wk: u32, ns: u64, ts: TimeScale) -> Self {
+        let week = Duration::from_seconds(wk as f64 * SECONDS_PER_DAY * 7.0);
+        Self::from_duration(week + (ns as f64) * Unit::Nanosecond, ts)
+    }
+
+    #[cfg(feature = "python")]
+    #[staticmethod]
+    pub fn from_timeofweek_utc(wk: u32, ns: u64) -> Self {
+        let week = Duration::from_seconds(wk as f64 * SECONDS_PER_DAY * 7.0);
+        Self::from_utc_duration(week + (ns as f64) * Unit::Nanosecond)
+    }
+
+    /// Converts to "time of week" (`wk, `tow`),
+    /// which is usually how GNSS receivers describe a timestamp.
+    /// `wk` is a rolling week conter into that time scale,
+    /// `tow` is the number of seconds since closest Sunday midnight into that week.
+    pub fn to_timeofweek_utc(&self) -> (u32, u64) {
+        // fractionnal days in this time scale
+        let days = match self.time_scale {
+            TimeScale::GPST => self.to_gpst_days(),
+            TimeScale::GST => self.to_gst_days(),
+            TimeScale::BDT => self.to_bdt_days(),
+            TimeScale::TT => self.to_tt_days(),
+            TimeScale::TAI => self.to_tai_days(),
+            TimeScale::UTC => self.to_utc_days(),
+            TimeScale::TDB => self.to_tdb_seconds() / SECONDS_PER_DAY,
+            TimeScale::ET => self.to_et_seconds() / SECONDS_PER_DAY,
+        };
+        // wk: rolling week counter into timescale
+        let wk = (days / 7.0) as u32;
+        // tow: number of nanoseconds between self and closest sunday midnight / start of week
+        let start_of_week = self.closest_utc_start_of_week();
+        let dw = *self - start_of_week; // difference in weekdays [0..6]
+        let (_, d, h, m, s, ms, us, ns) = dw.decompose();
+        let tow = Duration::from_days(d as f64)
+            + Duration::from_hours(h as f64)
+            + Duration::from_seconds((m * 60) as f64)
+            + Duration::from_seconds(s as f64)
+            + Duration::from_milliseconds(ms as f64)
+            + Duration::from_microseconds(us as f64)
+            + Duration::from_nanoseconds(ns as f64);
+        let (_, mut tow) = tow.to_parts();
+        tow += self.leap_seconds(true).unwrap_or(0.0) as u64 * 1_000_000_000;
+        (wk, tow)
+    }
+
+    /// Returns weekday counter in TAI timescale,
+    /// 0: Monday, ..., 6: Sunday
+    pub fn weekday_tai(&self) -> u8 {
```

Comment:
```suggestion
    pub fn weekday_tai(&self) -> Weekday {
```

And then create a new file called `weekday.rs` with the enum:

```rust
#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
pub enum Weekday {
    Monday,
    Tuesday,
    Wednesday,
    Thursday,
    Friday,
    Saturday,
    Sunday,
}
```

Reply:
This makes things already much simpler.